### PR TITLE
fix: SSHKey page nodes query error

### LINKED ISSUE #527: SSH keys page display cluster list when click refresh button
image

### PR DESCRIPTION
**Pull request by whDongRui**
resolve #527

### CODE REVIEW

File: src/pages/SSHKeys/index.jsx [hunk 1]
```diff
@@ -27,14 +27,17 @@ export default class SSHKeys extends Component {
   async componentDidMount() {
     const { clusterStore, clusterDetailStore, sshKeyStore, user } = this.props;
 
-    await sshKeyStore.fetchKeyPairs({ owner: user.user_id });
+    sshKeyStore.userId = user.user_id;
+    await sshKeyStore.fetchKeyPairs();
 
     const nodeIds = get(sshKeyStore.keyPairs[0], 'node_id', '');
+    clusterDetailStore.nodeIds = nodeIds || ['0'];
```
[the remark this answers is not in the document]
Comment (whDongRui) [reply src/pages/SSHKeys/index.jsx]: If nodeIds is  null will query all nodes data, set '0' for query no data.

File: src/pages/SSHKeys/index.jsx [hunk 2]
```diff
@@ -56,7 +59,8 @@ export default class SSHKeys extends Component {
     if (currentPairId !== item.key_pair_id) {
       cancelSelectNodes();
       sshKeyStore.currentPairId = item.key_pair_id;
-      fetchNodes({ node_id: item.node_id || 0 });
+      clusterDetailStore.nodeIds = item.node_id || ['0'];
+      fetchNodes();
```
[the remark this answers is not in the document]
Comment (whDongRui) [reply src/pages/SSHKeys/index.jsx]: ok, add await for fetchNodes method.

File: src/stores/key_pair/index.js [hunk 3]
```diff
@@ -37,7 +38,7 @@ export default class KeyPairStore extends Store {
     this.isLoading = true;
     const result = await this.request.get(
       'clusters/key_pairs',
-      Object.assign({ noLimit: true }, params)
+      Object.assign({ noLimit: true,  owner: this.userId }, params)
```
[the remark this answers is not in the document]
Comment (whDongRui) [reply src/stores/key_pair/index.js]: Refresh , query nodes and click left ssh key method all need userId.

File: src/pages/SSHKeys/index.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (iwisunny) [reply src/pages/SSHKeys/index.jsx]: OK

File: src/pages/SSHKeys/index.jsx [hunk 4]
```diff
@@ -27,14 +27,17 @@ export default class SSHKeys extends Component {
   async componentDidMount() {
     const { clusterStore, clusterDetailStore, sshKeyStore, user } = this.props;
 
-    await sshKeyStore.fetchKeyPairs({ owner: user.user_id });
+    sshKeyStore.userId = user.user_id;
```
[the remark this answers is not in the document]
Comment (whDongRui) [reply src/pages/SSHKeys/index.jsx]: Refresh , query nodes and click left ssh key methods need change, bacase there are need pass user_id parameter.

File: src/pages/SSHKeys/index.jsx [hunk 5]
```diff
@@ -27,14 +27,17 @@ export default class SSHKeys extends Component {
...
+    await sshKeyStore.fetchKeyPairs();
 
     const nodeIds = get(sshKeyStore.keyPairs[0], 'node_id', '');
+    clusterDetailStore.nodeIds = nodeIds || ['0'];
     if (nodeIds) {
       await clusterStore.fetchAll({
         noLimit: true,
         active: clusterStatus
       });
+      clusterDetailStore.nodeIds = nodeIds;
```
[the remark this answers is not in the document]
Comment (whDongRui) [reply src/pages/SSHKeys/index.jsx]: Refresh , query nodes  need pass nodeIds parameter.

File: src/stores/key_pair/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (iwisunny) [reply src/stores/key_pair/index.js]: OK， refactor it later

**Comment by iwisunny** [comment]
fix #527

**Pull request merged**